.github/workflows/deploy-gh-pages.yaml

```diff
@@ -21,8 +20,24 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  test-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "18"
+          cache: npm
+      - name: Install Dependencies
+        run: npm ci
+      - name: Test Build Next.js Project
+        run: npm run build
```

Comment:
Nice. However, the idea would be to run this "test build" on every Pull Request. Not the entire built job on every pull request, but just the check if it will build. Can you create a new Github Actuons `yaml` file, named `check-deploy.yaml` and and these steps there?

Reply:
@dmohns check again